framework/src/org/checkerframework/framework/flow/CFAbstractStore.java

```diff
@@ -980,69 +980,40 @@ public boolean equals(Object o) {
     @SideEffectFree
     @Override
     public String toString() {
-        return toDOToutput().replace("\\n", "\n");
+        return "Use a CFGVisualizer to see the Store: " + this.hashCode();
     }
 
-    @Pure
-    @Override
-    public boolean hasDOToutput() {
-        return true;
-    }
-
-    /**
-     * @return DOT representation of the store (may contain control characters
-     *         such as "\n").
-     */
+    @SuppressWarnings("unchecked")
     @Override
-    public String toDOToutput() {
-        StringBuilder result = new StringBuilder(this.getClass()
-                .getCanonicalName() + " (\\n");
-        internalDotOutput(result);
-        result.append(")");
-        return result.toString();
-    }
-
-
-    protected String escapeDoubleQuotes(final String str) {
-        return str.replace("\"", "\\\"");
-    }
-
-
-    protected String toStringEscapeDoubleQuotes(final Object obj) {
-        return escapeDoubleQuotes(String.valueOf(obj));
+	public void visualize(CFGVisualizer<?, S, ?> viz) {
+        ((CFGVisualizer<V, S, ?> ) viz).visualizeStoreHeader(
```

Comment:
Create a local variable instead of casting three times.
Document why the "unchecked" warning can be suppressed.